Dockerfile

```diff
@@ -58,10 +56,9 @@ RUN --mount=type=bind,target=. \
     golangci-lint run ./...
 
 FROM gobase AS version
-ARG PKG
 RUN --mount=target=. \
-    VERSION=$(git describe --match 'v[0-9]*' --dirty='.m' --always --tags); \
-    echo "-s -w -X ${PKG}/credentials.Version=${VERSION}" | tee /tmp/.ldflags; \
+    PKG=github.com/docker/docker-credential-helpers VERSION=$(git describe --match 'v[0-9]*' --dirty='.m' --always --tags) REVISION=$(git rev-parse HEAD)$(if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi); \
+    echo "-s -w -X ${PKG}/credentials.Version=${VERSION} -X ${PKG}/credentials.Revision=${REVISION} -X ${PKG}/credentials.Package=${PKG}" | tee /tmp/.ldflags; \
```

Comment:
Looks like we now won't be able to override these through build-args.

I think we at least need to be able to override "Package", but possibly also Revision and Version.